app/controllers/RestaurantController.java

```diff
@@ -81,6 +84,9 @@ public Result getAllRestaurants() {
 						.setPageSize(getQueryInt(request().getQueryString(PAGE_SIZE), DEFAULT_PAGE_SIZE))
 						.setNameFilter(request().getQueryString(NAME_FILTER))
 						.setCityFilter(!StringUtil.isNullOrEmpty(cityFilter) ? UUID.fromString(cityFilter) : null)
+                        .setPriceFilter(getQueryInt(request().getQueryString(PRICE_FILTER),0))
+						.setCuisineFilter(request().getQueryString(CUISINE_FILTER))
```

Comment:
Here you should use `String[] cuisineFilter = request().queryString().get(CUISINE_FILTER);` to get list of all cousines not only first one.